### PR TITLE
Added better error handling

### PR DESCRIPTION
**Pull request by BradleyCai**
Issue #76 

Also fixed issue #75 

Also added better css spacing

Rewrote a good bit of the js and moved a couple pieces around.

New error messages look like this:

image

### CODE REVIEW

File: regex/hours.regex [hunk 1]
```diff
@@ -1 +1 @@
-([0-9]{4}[AP]M)-([0-9]{4}[AP]M)
+([0-9]{4}[AP]M)?-([0-9]{4}[AP]M)?
```
Comment (emmiegit) [inline regex/hours.regex:1]: Why did you make each part of the time optional? This means that `-1200PM`, `0100PM-` and `-` are valid hours.

Comment (BradleyCai) [inline regex/hours.regex:1]: @ammongit 
It's a fix for issue #75 

A course like:

`NO TIMES`
`ENGR-092 -001    SN    1.00    TBA -`

Will get picked up, but it's dealt with later on in the program and displayed as an error.

Comment (BradleyCai) [inline regex/hours.regex:1]: @ammongit 
Added the case for a partially missing time like `-1200PM` and `0100PM-` in commit:

98791da7a4fe82e9cd1f734fcfa08445f87db737

**Comment by emmiegit** [comment]
The class `CONFLICTS TWICE` has two errors: shouldn't only one error per class be printed? Why does the first one only list two conflicts?

File: regex/hours.regex (hunk 1 above)
Comment (emmiegit) [inline regex/hours.regex:1]: The new regex matches `-` in `TBA` [like this](https://github.com/BradleyCai/ucr-schedule-visualizer/blob/issue-76/etc/test_all_errors.txt#L12). Nice.

**Comment by BradleyCai** [comment]
Error handling will be reworked with a code refactor in the future. 

It has been decided that this branch will not be merged, though the ideas of the issue will still be dealt with in the future.

**Pull request closed without merging**